### ISSUE TITLE
[Bug]: AttributeError: 'numpy.ndarray' object has no attribute 'to' in TorchInferencer

### ISSUE THREAD

**Issue by xingchengyusi** [openvinotoolkit/anomalib#2026]
### Describe the bug

When I use the TorchInferencer, it told me there are no attribute 'to' of the ndarray named 'image'. When I back to the source code, look like the problem is quite simple:

```python
    def pre_process(self, image: np.ndarray) -> torch.Tensor:
        """Pre process the input image.

        Args:
            image (np.ndarray): Input image

        Returns:
            Tensor: pre-processed image.
        """
        if len(image) == 3:
            image = image.unsqueeze(0)

        return image.to(self.device)
```

The input ndarray named 'image' is not a Tensor, and there should exists a convert sentence change a ndarray to Tensor.

### Dataset

Other (please specify in the text field below)

### Model

Other (please specify in the field below)

### Steps to reproduce the behavior

I use UCSDped as the dataset and the model is AI-VAD.

```python
inferencer = TorchInferencer(
    path=model_path,
    device='cuda',
)

# extract frames
file = Path(video_path)
assert file.is_file()

cap = cv2.VideoCapture(str(file))
success, image = cap.read()
scores = []
while success:
    s = inferencer.predict(image=image)
    scores.append(s)
    success, image = cap.read()

np.save('scores.npy', scores)
```

### OS information

OS information:
- OS: Ubuntu 20.04
- Python version: 3.10.14
- Anomalib version: 1.0.1
- PyTorch version: 2.1.0
- CUDA/cuDNN version: 11.3
- GPU models and configuration: GeForce RTX 4090
- Any other relevant information: In docker: nvidia-cuda-11.3.1-cudnn8-devel-ubuntu20.04

### Expected behavior

Normal Work and output a Tensor

### Pip/GitHub

pip

### What version/branch did you use?

1.0.1

### Configuration YAML

```yaml
name: anoma
channels:
  - pytorch
  - nvidia
  - defaults
dependencies:
  - _libgcc_mutex=0.1=main
  - _openmp_mutex=5.1=1_gnu
  - anyio=4.2.0=py310h06a4308_0
  - arrow=1.2.3=py310h06a4308_1
  - backoff=2.2.1=py310h06a4308_1
  - beautifulsoup4=4.12.2=py310h06a4308_0
  - blas=1.0=mkl
  - blessed=1.20.0=py310h06a4308_0
  - boto3=1.29.1=py310h06a4308_0
  - botocore=1.32.1=py310h06a4308_0
  - bottleneck=1.3.7=py310ha9d4c09_0
  - brotli-python=1.0.9=py310h6a678d5_7
  - bzip2=1.0.8=h5eee18b_5
  - ca-certificates=2024.3.11=h06a4308_0
  - certifi=2024.2.2=py310h06a4308_0
  - charset-normalizer=2.0.4=pyhd3eb1b0_0
  - click=8.1.7=py310h06a4308_0
  - croniter=1.3.7=py310h06a4308_0
  - cuda-cudart=11.8.89=0
  - cuda-cupti=11.8.87=0
  - cuda-libraries=11.8.0=0
  - cuda-nvrtc=11.8.89=0
  - cuda-nvtx=11.8.86=0
  - cuda-runtime=11.8.0=0
  - dateutils=0.6.12=py310h06a4308_0
  - deepdiff=6.7.1=py310h2f386ee_0
  - exceptiongroup=1.2.0=py310h06a4308_0
  - fastapi=0.103.0=py310h06a4308_0
  - ffmpeg=4.3=hf484d3e_0
  - filelock=3.13.1=py310h06a4308_0
  - freetype=2.12.1=h4a9f257_0
  - fsspec=2023.10.0=py310h06a4308_0
  - gmp=6.2.1=h295c915_3
  - gmpy2=2.1.2=py310heeb90bb_0
  - gnutls=3.6.15=he1e5248_0
  - h11=0.14.0=py310h06a4308_0
  - idna=3.4=py310h06a4308_0
  - inquirer=3.1.4=py310h06a4308_0
  - intel-openmp=2023.1.0=hdb19cb5_46306
  - itsdangerous=2.0.1=pyhd3eb1b0_0
  - jinja2=3.1.3=py310h06a4308_0
  - jmespath=1.0.1=py310h06a4308_0
  - joblib=1.2.0=py310h06a4308_0
  - jpeg=9e=h5eee18b_1
  - lame=3.100=h7b6447c_0
  - lcms2=2.12=h3be6417_0
  - ld_impl_linux-64=2.38=h1181459_1
  - lerc=3.0=h295c915_0
  - libcublas=11.11.3.6=0
  - libcufft=10.9.0.58=0
  - libcufile=1.9.1.3=0
  - libcurand=10.3.5.147=0
  - libcusolver=11.4.1.48=0
  - libcusparse=11.7.5.86=0
  - libdeflate=1.17=h5eee18b_1
  - libffi=3.4.4=h6a678d5_0
  - libgcc-ng=11.2.0=h1234567_1
  - libgfortran-ng=11.2.0=h00389a5_1
  - libgfortran5=11.2.0=h1234567_1
  - libgomp=11.2.0=h1234567_1
  - libiconv=1.16=h7f8727e_2
  - libidn2=2.3.4=h5eee18b_0
  - libjpeg-turbo=2.0.0=h9bf148f_0
  - libnpp=11.8.0.86=0
  - libnvjpeg=11.9.0.86=0
  - libpng=1.6.39=h5eee18b_0
  - libstdcxx-ng=11.2.0=h1234567_1
  - libtasn1=4.19.0=h5eee18b_0
  - libtiff=4.5.1=h6a678d5_0
  - libunistring=0.9.10=h27cfd23_0
  - libuuid=1.41.5=h5eee18b_0
  - libwebp-base=1.3.2=h5eee18b_0
  - lightning=2.0.9.post0=py310h06a4308_0
  - lightning-cloud=0.5.57=py310h06a4308_0
  - lightning-utilities=0.9.0=py310h06a4308_0
  - llvm-openmp=14.0.6=h9e868ea_0
  - lz4-c=1.9.4=h6a678d5_0
  - markupsafe=2.1.3=py310h5eee18b_0
  - mkl=2023.1.0=h213fc3f_46344
  - mkl-service=2.4.0=py310h5eee18b_1
  - mkl_fft=1.3.8=py310h5eee18b_0
  - mkl_random=1.2.4=py310hdb19cb5_0
  - mpc=1.1.0=h10f8cd9_1
  - mpfr=4.0.2=hb69a4c5_1
  - mpmath=1.3.0=py310h06a4308_0
  - ncurses=6.4=h6a678d5_0
  - nettle=3.7.3=hbbd107a_1
  - networkx=3.1=py310h06a4308_0
  - numexpr=2.8.7=py310h85018f9_0
  - numpy=1.26.4=py310h5f9d8c6_0
  - numpy-base=1.26.4=py310hb5e798b_0
  - openh264=2.1.1=h4ff587b_0
  - openjpeg=2.4.0=h3ad879b_0
  - openssl=3.0.13=h7f8727e_0
  - ordered-set=4.1.0=py310h06a4308_0
  - orjson=3.9.15=py310h97a8848_0
  - packaging=23.2=py310h06a4308_0
  - pandas=2.2.1=py310h6a678d5_0
  - pillow=10.2.0=py310h5eee18b_0
  - pip=23.3.1=py310h06a4308_0
  - psutil=5.9.0=py310h5eee18b_0
  - pyjwt=2.4.0=py310h06a4308_0
  - pysocks=1.7.1=py310h06a4308_0
  - python=3.10.14=h955ad1f_0
  - python-dateutil=2.8.2=pyhd3eb1b0_0
  - python-editor=1.0.4=pyhd3eb1b0_0
  - python-tzdata=2023.3=pyhd3eb1b0_0
  - pytorch=2.1.0=py3.10_cuda11.8_cudnn8.7.0_0
  - pytorch-cuda=11.8=h7e8668a_5
  - pytorch-lightning=2.0.3=py310h06a4308_0
  - pytorch-mutex=1.0=cuda
  - pytz=2023.3.post1=py310h06a4308_0
  - pyyaml=6.0.1=py310h5eee18b_0
  - readchar=4.0.5=py310h06a4308_0
  - readline=8.2=h5eee18b_0
  - requests=2.31.0=py310h06a4308_1
  - s3transfer=0.7.0=py310h06a4308_0
  - scikit-learn=1.3.0=py310h1128e8f_1
  - setuptools=68.2.2=py310h06a4308_0
  - six=1.16.0=pyhd3eb1b0_1
  - sniffio=1.3.0=py310h06a4308_0
  - soupsieve=2.5=py310h06a4308_0
  - sqlite=3.41.2=h5eee18b_0
  - starlette=0.27.0=py310h06a4308_0
  - starsessions=1.3.0=py310h06a4308_0
  - sympy=1.12=py310h06a4308_0
  - tbb=2021.8.0=hdb19cb5_0
  - threadpoolctl=2.2.0=pyh0d69192_0
  - tk=8.6.12=h1ccaba5_0
  - torchaudio=2.1.0=py310_cu118
  - torchmetrics=1.1.2=py310h06a4308_0
  - torchtriton=2.1.0=py310
  - torchvision=0.16.0=py310_cu118
  - tqdm=4.65.0=py310h2f386ee_0
  - traitlets=5.7.1=py310h06a4308_0
  - typing-extensions=4.9.0=py310h06a4308_1
  - typing_extensions=4.9.0=py310h06a4308_1
  - tzdata=2024a=h04d1e81_0
  - urllib3=2.0.7=py310h06a4308_0
  - uvicorn=0.20.0=py310h06a4308_0
  - websocket-client=0.58.0=py310h06a4308_4
  - websockets=10.4=py310h5eee18b_1
  - wheel=0.41.2=py310h06a4308_0
  - xz=5.4.6=h5eee18b_0
  - yaml=0.2.5=h7b6447c_0
  - zlib=1.2.13=h5eee18b_0
  - zstd=1.5.5=hc292b87_0
  - pip:
      - aiofiles==23.2.1
      - altair==5.3.0
      - annotated-types==0.6.0
      - anomalib==1.0.1
      - antlr4-python3-runtime==4.9.3
      - attrs==23.2.0
      - av==12.0.0
      - contourpy==1.2.1
      - cycler==0.12.1
      - docstring-parser==0.16
      - einops==0.7.0
      - ffmpy==0.3.2
      - fonttools==4.51.0
      - freia==0.2
      - ftfy==6.2.0
      - gradio==4.27.0
      - gradio-client==0.15.1
      - httpcore==1.0.5
      - httpx==0.27.0
      - huggingface-hub==0.22.2
      - imageio==2.34.1
      - imgaug==0.4.0
      - importlib-resources==6.4.0
      - jsonargparse==4.27.1
      - jsonschema==4.21.1
      - jsonschema-specifications==2023.12.1
      - kiwisolver==1.4.5
      - kornia==0.7.2
      - kornia-rs==0.1.3
      - lazy-loader==0.4
      - markdown-it-py==3.0.0
      - matplotlib==3.8.4
      - mdurl==0.1.2
      - objprint==0.2.3
      - omegaconf==2.3.0
      - open-clip-torch==2.24.0
      - opencv-python==4.9.0.80
      - protobuf==5.26.1
      - pydantic==2.7.1
      - pydantic-core==2.18.2
      - pydub==0.25.1
      - pygments==2.17.2
      - pyparsing==3.1.2
      - python-multipart==0.0.9
      - referencing==0.34.0
      - regex==2024.4.16
      - rich==13.7.1
      - rich-argparse==1.4.0
      - rpds-py==0.18.0
      - ruff==0.4.1
      - safetensors==0.4.3
      - scikit-image==0.23.2
      - scipy==1.13.0
      - semantic-version==2.10.0
      - sentencepiece==0.2.0
      - shapely==2.0.4
      - shellingham==1.5.4
      - tifffile==2024.4.18
      - timm==0.9.16
      - tomlkit==0.12.0
      - toolz==0.12.1
      - typer==0.12.3
      - viztracer==0.16.2
      - wcwidth==0.2.13
prefix: /root/miniconda3/envs/anoma
```

### Logs

```shell
Could not find openvino. To use this feature, ensure that you have openvino installed.
OpenVINO is not installed. Please install OpenVINO to use OpenVINOInferencer.
INFO:root:pretrained_model/weights/torch/model.pt
Traceback (most recent call last):
  File "/root/anomalib_aivad/inference.py", line 46, in <module>
    main('/root/anomaly_test/anomaly_j3.mp4')
  File "/root/anomalib_aivad/inference.py", line 39, in main
    s = inferencer.predict(image=image)
  File "/root/miniconda3/envs/anoma/lib/python3.10/site-packages/anomalib/deploy/inferencers/torch_inferencer.py", line 187, in predict
    processed_image = self.pre_process(image)
  File "/root/miniconda3/envs/anoma/lib/python3.10/site-packages/anomalib/deploy/inferencers/torch_inferencer.py", line 213, in pre_process
    return image.to(self.device)
AttributeError: 'numpy.ndarray' object has no attribute 'to'
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by xingchengyusi** [comment]
And I want to take it~

**Comment by TurboJonte** [comment]
There's a PR already for this:
https://github.com/openvinotoolkit/anomalib/pull/1816

**Comment by xingchengyusi** [comment]
Thank you, I only check the issue to make sure no duplicate.

**Comment by blueclowd** [comment]
May I know if it is solved on anomalib v1.0.1?
I am having this issue on anomalib v1.0.1 installed using PIP
